packages/core/src/operator/operator-runtime/validateConfirmData.ts

```diff
@@ -0,0 +1,144 @@
+import { operatorState } from "./operatorState.js";
+import { Challenge, config, getConfirmDataAndHash, PublicNodeBucketInformation, verifyChallengerSignedHash } from "../../index.js";
+import axios from "axios";
+import { ethers } from "ethers";
+
+
+export async function validateConfirmData(currentChallenge: Challenge, subgraphIsHealthy: boolean, event?: ethers.EventLog): Promise<boolean> {
+
+    try {
+        if (event && currentChallenge.rollupUsed === config.rollupAddress) {
+
+            const currentAssertionId = Number(currentChallenge.assertionId);                // Destructure Current Assertion ID     
+            let assertionIds: number[] = [];                                                // Create an array to store the assertion IDs  
+
+            // Loop through the assertion IDs and add them to the array
+            for (let id = Number(operatorState.previousChallengeAssertionId) + 1; id <= currentAssertionId; id++) {
+                assertionIds.push(id);
+            }
+
+            const isBatch = assertionIds.length > 1;                                        // Check if the challenge is a batch or single challenge        
+            let confirmDataList: string[] = [];                                             // Create an array to store the confirm data for each assertionId                          
+
+            if (isBatch) {
+                const { confirmData } = await getConfirmDataAndHash(assertionIds, subgraphIsHealthy);
+                confirmDataList = confirmData;                                              // Set the confirm data list        
+            } else {
+                confirmDataList = [currentChallenge.assertionStateRootOrConfirmData];       // Set the initial confirm data assuming a single challenge
+            }
+
+            const errors = [];
+
+            // Validate the confirm data for each assertionId
+            for (let i = 0; i < confirmDataList.length; i++) {
+                const confirmData = confirmDataList[i];
+                const assertionId = assertionIds[i];
+
+                try {
+                    const { publicNodeBucket, error } = await compareWithCDN(assertionId, confirmData);
+
+                    if (error) {
+                        errors.push({ assertionId, error });
+                        operatorState.onAssertionMissMatchCb(publicNodeBucket, currentChallenge, error);
```

Comment:
We should log all errors together and not trigger `onAssertionMissMatchCb` multiple times.
Seems we already push to the error array for that reason.

Reply:
This has been revised, thank you.